### PR TITLE
Move from boost to c++11

### PR DESCRIPTION
**Pull request by patriotyk**

### CODE REVIEW

File: src/httpclient.cc [hunk 1]
```diff
@@ -96,21 +95,21 @@ void HttpClient::setCerts(const std::string& ca, CryptoSource ca_source, const s
   if (ca_source == kPkcs11) {
     throw std::runtime_error("Accessing CA certificate on PKCS11 devices isn't currently supported");
   } else {  // ca_source =kFile
-    boost::movelib::unique_ptr<TemporaryFile> tmp_ca_file = boost::movelib::make_unique<TemporaryFile>("tls-ca");
+    std::unique_ptr<TemporaryFile> tmp_ca_file = std::make_unique<TemporaryFile>("tls-ca");
```
Comment (patrickvacek) [inline src/httpclient.cc]: I thought `make_unique` was C++14, which I don't think we're authorized to use yet.

Comment (patrickvacek) [reply src/httpclient.cc]: Oh, now I see! You've made your own version in the meantime. That's fine with me for now, but any particular reason to prefer that over the boost version we've been using?

**Comment by patrickvacek** [comment]
I found a few more related things:

```
$ grep BOOST_FOREACH src/ tests/ -rIn
src/sota_tools/ostree_dir_repo.cc:57:  BOOST_FOREACH (std::string ext, exts) {
src/sota_tools/ostree_http_repo.cc:74:  BOOST_FOREACH (std::string ext, exts) {
$ grep "boost::noncopyable" src/ tests/ -rIn
src/utils.h:45:class TemporaryFile : boost::noncopyable {
src/p11engine.h:68:class P11Engine : public boost::noncopyable {
src/utils.cc:549:class SafeTempRoot : private boost::noncopyable {
$ grep "boost/noncopyable" src/ tests/ -rIn
src/sota_tools/ostree_dir_repo.h:6:#include <boost/noncopyable.hpp>
$ grep "move/unique_ptr" src/ tests/ -rIn
src/keymanager.h:4:#include <boost/move/unique_ptr.hpp>
$ grep "boost/random" src/ tests/ -rIn
tests/opcuabridge_messaging_test.cc:13:#include <boost/random/mersenne_twister.hpp>
tests/opcuabridge_messaging_test.cc:14:#include <boost/random/uniform_int.hpp>
tests/opcuabridge_secondary_update_test.cc:18:#include <boost/random/mersenne_twister.hpp>
tests/opcuabridge_secondary_update_test.cc:19:#include <boost/random/uniform_int.hpp>
tests/utils_test.cc:12:#include <boost/random/mersenne_twister.hpp>
tests/utils_test.cc:13:#include <boost/random/uniform_int_distribution.hpp>
tests/utils_test.cc:14:#include <boost/random/uniform_smallint.hpp>
$ grep "boost::random" src/ tests/ -rIn
tests/opcuabridge_messaging_test.cc:56:  boost::random::mt19937 gen;
tests/opcuabridge_messaging_test.cc:58:  boost::random::uniform_int_distribution<uint8_t> random_byte(0x00, 0xFF);
tests/opcuabridge_secondary_update_test.cc:38:  boost::random::mt19937 gen;
tests/opcuabridge_secondary_update_test.cc:40:  boost::random::uniform_int_distribution<uint8_t> random_byte(0x00, 0xFF);
tests/utils_test.cc:85:  boost::random::uniform_smallint<char> chars(std::numeric_limits<char>::min(), std::numeric_limits<char>::max());
tests/utils_test.cc:87:  boost::random::uniform_smallint<int> length(0, 20);
$ grep "boost/bind" src/ tests/ -rIn
src/opcuabridge/opcuabridgeserver.cc:8:#include <boost/bind.hpp>
$ grep "boost::bind" src/ tests/ -rIn
src/sotauptaneclient.cc:137:  std::thread polling_thread(boost::bind(&SotaUptaneClient::run, this, commands_channel));
src/opcuabridge/opcuabridgeserver.cc:29:  model_->file_list_.setOnAfterWriteCallback(boost::bind(&Server::onFileListUpdated, this, _1));
src/opcuabridge/opcuabridgeserver.cc:30:  model_->metadatafile_.setOnAfterWriteCallback(boost::bind(&Server::countReceivedMetadataFile, this, _1));
src/opcuabridge/opcuabridgeserver.cc:31:  model_->version_report_.setOnBeforeReadCallback(boost::bind(&Server::onVersionReportRequested, this, _1));
```

**Comment by patrickvacek** [comment]
There's also this comment that I made months ago:

`src/sotauptaneclient.cc:252:  // Should be a const_iterator but we need C++11 for cbegin.`

I seem to recall many places where I longed for constant iterators but had to do without. I'd also love to see constexpr used anywhere that it makes sense (which might be nowhere).

File: src/httpclient.cc (hunk 1 above)
Comment (patriotyk) [reply src/httpclient.cc]: I thought it is in c++11 but then I have googled implementation(https://isocpp.org/files/papers/N3656.txt) that is accepted for c++14

**Comment by patriotyk** [comment]
Thank you, for more places with boost code, but opcuabridge I didnot replace because I don't know how to build and test it and I am not sure is it covered with tests. But I can do it if You think it is ok.

**Comment by patriotyk** [comment]
Hm, strange, I renmeber I removed all boost::noncopyable  places

**Comment by patrickvacek** [comment]
@patriotyk Just to be careful, you can probably leave the opcuabridge parts alone for now.

@riqkum Would you mind fixing those unnecessary boost references in the opcuabridge code when you get a chance? We do support C++11 now, so we should use the standard library versions wherever possible.

File: src/sotauptaneclient.h [hunk 2]
```diff
@@ -44,13 +42,14 @@ class SotaUptaneClient {
...
-  const boost::shared_ptr<INvStorage> storage;
-  boost::shared_ptr<PackageManagerInterface> pacman;
+  const std::shared_ptr<INvStorage> storage;
+  std::shared_ptr<PackageManagerInterface> pacman;
   HttpInterface &http;
   int last_targets_version;
   Json::Value operation_result;
   std::unique_ptr<IpUptaneConnection> ip_uptane_connection;
   std::unique_ptr<IpUptaneConnectionSplitter> ip_uptane_splitter;
+  bool shutdown = {false};
```
Comment (lbonn) [inline src/sotauptaneclient.h]: `std::atomic<bool> shutdown{false};` should probably be used for this kind of inter-thread signaling

File: src/utils.h [hunk 3]
```diff
@@ -84,4 +88,36 @@ struct SocketCloser {
 using SocketHandle = std::unique_ptr<int, SocketCloser>;
 bool operator<(const sockaddr_storage &left, const sockaddr_storage &right);  // required by std::map
```
Comment (lbonn) [inline src/utils.h]: great!

**Pull request merged**